### PR TITLE
Style improvements to Flask sample

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This PR improves the style of the Python code in the Flask+PG sample to be more Pythonic. I ran `isort`, `ruff`, `black --skip-string-normalization --line-length 200`, and `pyupgrade`, and then kept the most significant/objectively Pythonic changes. I did not keep any changes that would add/subtract new lines, since the articles reference line numbers, and such changes aren't significant anyway.

I will add inline comments to point out the kind of changes made.

## Does this introduce a breaking change?
```
[ ] Yes
[C] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[X] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Open in Codespaces
* azd up
* Add a restaurant

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
-from datetime import datetime
 import os
+from datetime import datetime
 
-from flask import Flask, render_template, request, redirect, url_for, send_from_directory
-from flask_sqlalchemy import SQLAlchemy
+from flask import Flask, redirect, render_template, request, send_from_directory, url_for
```
Comment (pamelafox) [inline app.py:4]: Import re-ordering comes from isort

File: app.py [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 
 app = Flask(__name__, static_folder='static')
 csrf = CSRFProtect(app)
 
 # WEBSITE_HOSTNAME exists only in production environment
-if not 'WEBSITE_HOSTNAME' in os.environ:
-   # local development, where we'll use environment variables
-   print("Loading config.development and environment variables from .env file.")
-   app.config.from_object('azureproject.development')
+if 'WEBSITE_HOSTNAME' not in os.environ:
```
Comment (pamelafox) [inline app.py:14]: Ruff recommends "not in" versus "not X in" due to it resulting in different true/false values in some situations, I believe.

File: app.py [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+    # local development, where we'll use environment variables
```
Comment (pamelafox) [inline app.py:15]: These lines and others were indented with 3 spaces instead of 4.

File: app.py [hunk 4]
```diff
@@ -37,13 +37,13 @@
 @app.route('/', methods=['GET'])
 def index():
     print('Request for index page received')
-    restaurants = Restaurant.query.all()
```
Comment (pamelafox) [inline app.py:40]: Removed trailing whitespace here and elsewhere

File: app.py [hunk 5]
```diff
@@ -37,13 +37,13 @@
...
     print('Request for index page received')
-    restaurants = Restaurant.query.all()    
+    restaurants = Restaurant.query.all()
     return render_template('index.html', restaurants=restaurants)
 
 @app.route('/<int:id>', methods=['GET'])
 def details(id):
     restaurant = Restaurant.query.where(Restaurant.id == id).first()
-    reviews = Review.query.where(Review.restaurant==id)
+    reviews = Review.query.where(Review.restaurant == id)
```
Comment (pamelafox) [inline app.py:46]: Black adds spaces around operators (notice that line 45 already had that, so this change also increases internal consistency)

File: models.py [hunk 6]
```diff
@@ -28,4 +29,4 @@ def validate_rating(self, key, value):
         return value
 
     def __str__(self):
-        return self.restaurant.name + " (" + self.review_date.strftime("%x") +")"
+        return f"{self.restaurant.name} ({self.review.date:%x})"
```
Comment (pamelafox) [inline models.py:32]: Generally f-strings are preferred when the expressions inside are fairly simple, since they're more readable and less error prone.

**Comment by vmagelo** [comment]
Thank you!

**Pull request merged**